dev_guide/application_health.adoc

```diff
@@ -11,33 +11,126 @@ toc::[]
+Set a readiness check by configuring the
+`*template.spec.containers.readinessprobe*` stanza of a pod configuration.
 
-OpenShift users can specify high-level health checks and control:
+|===
 
-<1> The health check methodology (viz: specify a script to run)
-<2> What action should be taken when a health check has a negative result (viz: create an Event)
+Both probes can be configured in three ways:
+
+*HTTP Checks*
+
+The kubelet uses a web hook to determine the healthiness of the container. The
+check is deemed successful if the hook returns with 200 or 399. The following is
+an example of a readiness check using the HTTP checks method:
+
+.Readiness HTTP check
+====
+----
+...
+readinessProbe:
+  httpGet:
+    path: /healthz
+    port: 8080
+  initialDelaySeconds: 15
+  timeoutSeconds: 1
+...
+----
+====
+
+A HTTP check is ideal for complex applications that can return with a 200 status
+when completely initialized.
+
+*Container Execution Checks*
+
+The kubelet executes a command inside the container. Exiting the check with
+status 0 is considered a success. The following is an example of a liveness
+check using the container execution method:
+
+.Liveness Container Execution Check
+====
+----
+...
+livenessProbe:
+  exec:
+    command:
+    - cat
+    - /tmp/health
+  initialDelaySeconds: 15
+  timeoutSeconds: 1
+...
+----
+====
+
+*TCP Socket Checks*
+
+The kubelet attempts to open a socket to the container. The container is only
+considered healthy if the check can establish a connection. The following is an
+example of a liveness check using the TCP socket check method:
+
+.Liveness TCP Socket Check
+====
+----
+...
+livenessProbe:
+  tcpSocket:
+    port: 8080
+  initialDelaySeconds: 15
+  timeoutSeconds: 1
+...
+----
+====
+
+A TCP socket check is ideal for applications that do not start listening until
+initialization is complete.
+
+For more information on health checks, see the
+http://kubernetes.io/v1.0/docs/user-guide/walkthrough/k8s201.html#health-checking[Kubernetes
+documentation].
+
+[[high-level-application-health-checks]]
+
+== High-level Application Health Checks
+
+Holistic notions of application health that cut across multiple components
+cannot be handled with container-scoped liveness checks. Having healthy and
+responsive system components in a distributed application, but in a system with
+overall unhealthy behavior, is possible. Administrators of distributed systems
+use health checks to test the behavior of the overall health of a system.
+
+A typical example is a script that performs a test transaction, exercising the
+components of the system together and verifying that the high-level system
+behavior in the context of the test transaction matches the expected behavior of
+a health system.
+
+OpenShift users can specify high-level health checks and control the health
+check methodology by specifying a script to run, and the action to be taken when
+a health check has a negative result by creating an event.
```

Comment:
The answer to all of these questions is... I don't know. Because this issue was about the health checks and using the info from the Kubernetes doc to fix that up, I'll make another about this section and ask for examples, etc.